report/report.md

```diff
@@ -0,0 +1,898 @@
+as this concept lends itself well to the structure of blockchains. This
+allows a blockchain node to omit all previous transaction history in
+favour of only a single state, for example, containing all current account
+balances. This is a so-called _succinct blockchain_, one such blockchain is
+[Mina](https://minaprotocol.com/).
+
+In order to acheive IVC, you need a function $F \in S \to S$ along with some
+initial state $z_0 \in S$. Then you can call $F$ to generate a series of
+$z$'s, $\vec{z} \in S^{n+1}$:
+
+\begin{figure}[!H]
+\centering
+\begin{tikzpicture}[node distance=2cm]
+
+    % Nodes
+    \node (z0) [node] {$z_0$};
+    \node (z1) [node, right=of z0] {$z_1$};
+    \node (z2) [node, right=of z1] {$z_2$};
+    \node (dots) [right=1cm of z2] {$\dots$};
+    \node (zn) [node, right=1cm of dots] {$z_n$};
+
+    % Arrows with labels
+    \draw[arrow] (z0) -- node[above] {$F$} (z1);
+    \draw[arrow] (z1) -- node[above] {$F$} (z2);
+    \draw[arrow] (z2) -- node[above] {$F$} (dots);
+    \draw[arrow] (dots) -- node[above] {$F$} (zn);
+
+\end{tikzpicture}
+\caption{A visualization of the relationship between $F$ and $\vec{z}$ in a non-IVC setting.}
+\end{figure}
+
+In a blockchain setting, you might imagine any $z_i \in \vec{z}$ as a set
+of accounts with corresponding balances, and the transition function $F$
+as the computation happening when a new block is created and therefore
+a new state $z_i$ In the IVC setting, we have a proof, $\pi$, associated
+with each state, so that anyone can take just a single pair $(z_m, \pi_m)$
+along with the initial state and transition function ($z_0, F$) and verify
+that said state was computed correctly.
+
+\begin{figure}[!H]
+\centering
+\begin{tikzpicture}[node distance=2cm]
+
+    % Nodes
+    \node (z0) [node] {$z_0$};
+    \node (z1) [node, right=of z0] {$(z_1, \pi_1)$};
+    \node (z2) [node, right=of z1] {$(z_2, \pi_2)$};
+    \node (dots) [right=1cm of z2] {$\dots$};
+    \node (zn) [node, right=1cm of dots] {$(z_n, \pi_n)$};
+
+    % Arrows with labels
+    \draw[arrow] (z0) -- node[above] {$F$} (z1);
+    \draw[arrow] (z1) -- node[above] {$F$} (z2);
+    \draw[arrow] (z2) -- node[above] {$F$} (dots);
+    \draw[arrow] (dots) -- node[above] {$F$} (zn);
+
+\end{tikzpicture}
+\caption{A visualization of the relationship between $F, \vec{z}$ and $\vec{\pi}$ in an IVC setting.}
+\end{figure}
+
+The proof $\pi_i$ describes the following:
+
+> _\textcolor{GbFg3}{"The current state $z_i$ is computed from applying $F$ $i$ times to $z_0$
+> ($z_i = F^i(z_0) = F(z_{i-1})$) and the associated proof $\pi_{i-1}$ for
+> the previous state is valid."}_
+
+Or more formally, $\pi_i$ is a proof of the following claim expressed as a
+circuit $R$:
+
+$$z_i = F(z_{i-1}) \; \land \; (i = 0 \lor \exists \, \pi_{i-1}, \, \text{ s.t. } \SNARKVerifier(z_{i-1}, \pi_{i-1}) = \top)$$
+
+Where $V$ represents the verification circuit in the proof system we're
```

Comment:
What/where is V?

Reply:
Yes, my bad. I changed `v` to `\SNARKVerifier` at some point for clarity. Capturing different verifiers transparently, was somewhat of a challenge, so I decided to make it more explicit.